docs/architecture/index.md

```diff
@@ -0,0 +1,130 @@
+# Architecture Overview
+
+## Introduction
+
+### Which problem Zentral solves?
+**Asset Management**: helps organizations in managing their IT assets, tracking hardware and software inventory, and monitoring device health and performance.
+
+**Endpoint Security**: provides an all-in-one solution for endpoint security that includes continuous monitoring, vulnerability scanning, and malware detection.
+
+**Threat Detection**: helps in identifying and responding to advanced threats, malware, and suspicious activities through its advanced threat detection features, including behavior-based analytics, network traffic analysis, and log monitoring.
+Incident Response: provides a robust incident response system that enables teams to investigate security incidents and respond to them effectively.
+
+**Compliance**: assists in maintaining regulatory compliance by providing automated auditing, reporting, and compliance management features.
+
+### Which are the key benefits?
+**Centralized Endpoint Management**: offers centralized management of endpoints, making it easier for organizations to monitor and manage their IT assets effectively.
+
+**Advanced Threat Detection**: advanced threat detection features enable organizations to detect and respond to advanced threats, malware, and suspicious activities in real-time.
+
+**Automate Incident Response**: provides a powerful incident response system that enables organizations to automate and streamline their incident response processes, reducing response times and minimizing the impact of security incidents.
+Comprehensive Compliance Management: helps organizations stay compliant with various regulations and standards by providing automated auditing, reporting, and compliance management features.
+
+**Open Source Platform**: is an open-source platform, which means it's highly customizable and can be extended with custom plugins and integrations.
+
+**Low Cost of Ownership**: is free to use and provides a low cost of ownership, making it an affordable solution for organizations of all sizes.
+
+
+## Architecture overview
+
+### Key Components
+
+![Software Architecture overview](/images/architecture/key_components.svg)
+
+#### Description
+
+##### Endpoint
+
+**Agents**: supports agents that can be installed on endpoints to collect and send data to the Zentral Server for analysis and processing.
+
+**Enrollment profile**: a configuration that enrolls devices, into an organization's endpoint management system.
+
+**Sensors**: collect endpoint data, including hardware and software inventory, running processes, and network connections.
+
+##### Zentral
+
+**Zentral Server**: is the core component of the platform that manages device registration, data ingestion, and data processing.
+
+**Workers**: to perform background tasks, such as data ingestion, device registration, and alerting, including Celery and RabbitMQ is being used.
```

Comment:
Instead of device registration, maybe "event enrichment".

Celery is a weird one here, because it is not really part of the Zentral contrib module worker framework. It is just a worker that is needed for some long running tasks, like the exports.

RabbitMQ is the queue, the workers are agnostic to the queue backend.

Reply:
Dropping ", including Celery and RabbitMQ is being used." could be a solution.